front-end/src/components/EventCard.js

```diff
@@ -14,41 +16,69 @@ class EventCard extends Component {
   }
 
   handleExpandChange = expanded => this.setState({expanded: expanded});
+  onMouseOver = () => this.setState({ shadow: 2 });
+  onMouseOut = () => this.setState({ shadow: 1 });
+
+  renderLinks = (attachment, i) => {
+    const children = (
+      <div className="LinkChildren">
+        <a href={attachment.link}>{attachment.title}</a>
+        <img src={attachment.imageURL} alt={i}/>
+      </div>
+    );
+    return <CardText className="Link" key={i} expandable={true} children={children}></CardText>;
+  }
+
+  renderVideos = (attachment, i) => {
+    const children = <ReactPlayer className="Video" url={attachment.videoURL}/>;
+    return <CardMedia key={i} expandable={true} children={children}></CardMedia>;
+  }
+
+  renderImages = (attachment, i) => (
+    <CardMedia key={i} expandable={true}>
+      <img src={attachment.imageURL} alt={i}/>
+    </CardMedia>
+  )
+
+  renderTweets = (attachment, i) => {
+    let tweetID = attachment.tweetURL.split('/');
+    tweetID = tweetID.pop();
+    const children = <TweetEmbed id={tweetID} />;
+    return <CardMedia key={i} expandable={true} children={children}></CardMedia>;
+  }
 
   renderAttachments = () => {
```

Comment:
It can be worth refactoring this into `<Attachment  data={} />`. Then this component will be the one that decides how to render it. Just a comment, not a request for this PR.